martian/src/stage.rs

```diff
@@ -499,8 +502,65 @@ pub trait MartianStage: MroMaker {
         println!(" > [stage ] complete");
         result
     }
+
+    #[cfg(feature = "rayon")]
+    fn test_run(
+        &self,
+        run_directory: impl AsRef<Path> + Send + Sync,
+        args: Self::StageInputs,
+    ) -> Result<Self::StageOutputs, Error>
+    where
+        Self: Sync,
+        Self::ChunkInputs: Clone + Send + Sync,
+        Self::StageInputs: Clone + Send + Sync,
+        Self::ChunkOutputs: Send + Sync,
+    {
+        // Use default resource for split
+        let default_resource = Resource::new().mem_gb(1).vmem_gb(2).threads(1);
+        let split_path = prep_path(run_directory.as_ref(), "split")?;
+        let rover = MartianRover::new(split_path, default_resource);
+        println!("{}", vec!["-"; 80].join(""));
+        println!("{}", Self::stage_name());
+        println!("{}", vec!["-"; 80].join(""));
+        println!(" > [split ] running");
+        let stage_defs = self.split(args.clone(), rover)?;
+        println!(" > [split ] complete");
+
+        let run_chunk = |chunk: &ChunkDef<Self::ChunkInputs>,
+                         chunk_idx: usize|
+         -> Result<Self::ChunkOutputs, Error> {
+            println!(" > [chunk ] running with rayon {}", chunk_idx,);
+            let chunk_path = prep_path(run_directory.as_ref(), &format!("chnk{}", chunk_idx))?;
+            let rover = MartianRover::new(chunk_path, fill_defaults(chunk.resource));
+            self.main(args.clone(), chunk.inputs.clone(), rover)
+        };
+
+        println!(" > [chunks] {} chunks in total", stage_defs.chunks.len());
+        let chunk_outs = stage_defs
+            .chunks
+            .par_iter()
```

Comment:
@pmarks  This function is essentially a copy of the non-rayon version with added constraints on various types and this line changed from `iter` to `par_iter`

Reply:
if the chunks all requests 4 threads and rayon makes 1 thread per CPU this could result in a lot of threads... maybe that's OK for testing purposes?

Reply:
Yeah, I think it's OK for testing.